src/compiler/ruby_generator_string-inl.h

```diff
@@ -81,13 +81,29 @@ inline bool ReplacePrefix(grpc::string* s, const grpc::string& from,
   return true;
 }
 
-// CapitalizeFirst capitalizes the first char in a string.
-inline grpc::string CapitalizeFirst(grpc::string s) {
+// Modularize converts a string into a ruby module compatible name
+inline grpc::string Modularize(grpc::string s) {
   if (s.empty()) {
     return s;
   }
-  s[0] = ::toupper(s[0]);
-  return s;
+
+  grpc::string new_string = "";
+
+  bool was_last_underscore = false;
+
+  for (grpc::string::size_type i = 0; i < s.size(); ++i) {
+    if (i == 0 && s[i] != '_') {
```

Comment:
nit: I think we can slightly simplify this. Perhaps remove the first `if` here, and set `s[0] = ::toupper(s[0]);` at the top of this function (this shouldn't need to handle the case of `_` being the first character of the string).

Also, style nit: please remove the blank lines in this function